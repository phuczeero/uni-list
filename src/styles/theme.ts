const fonts = {
  body:
    '-apple-system, BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji"',
};
const fontSizes = [12, 14, 16, 20, 24, 32, 48, 64, 72];
const space = [0, 4, 8, 16, 32, 64, 128, 256, 512];
const radii = {
  default: 6,
  circle: 99999,
};
const colors = {
  text: '#212529',
  muted: '#dee2e6',
  primary: '#007bff',
  danger: '#dc3545',
  blues: {
    light: '#dcedff',
    dark: '#004896',
  },
};
const sizes = {
  navbar: 64,
  spinner: 32,
};
const shadows = {
  card: '0 0 4px rgba(0, 0, 0, .125)',
};
const breakpoints = ['576px', '768px', '992px', '1200px'];

export const theme = {
  fonts,
  fontSizes,
  space,
  radii,
  colors,
  sizes,
  shadows,
  breakpoints,
};

export type ThemeType = typeof theme;
