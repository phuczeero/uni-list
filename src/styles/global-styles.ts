import { createGlobalStyle } from 'styled-components';
import { ThemeType } from './theme';

export const GlobalStyle = createGlobalStyle<{ theme: ThemeType }>`
  * {
    font-family: ${p => p.theme.fonts.body};
    color: ${p => p.theme.colors.text}
  }
`;
