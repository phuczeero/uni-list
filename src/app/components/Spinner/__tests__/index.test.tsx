import * as React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components/macro';

import { theme } from 'styles/theme';
import { Spinner } from '..';

const renderWithTheme = () =>
  render(
    <ThemeProvider theme={theme}>
      <Spinner />
    </ThemeProvider>,
  );

describe('<Spinner  />', () => {
  it('should match snapshot', () => {
    const component = renderWithTheme();
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
