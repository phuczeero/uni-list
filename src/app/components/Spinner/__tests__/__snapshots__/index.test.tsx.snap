// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner  /> should match snapshot 1`] = `
.c0 {
  -webkit-animation: cqkDSr 2.625s linear infinite;
  animation: cqkDSr 2.625s linear infinite;
  height: 32px;
  width: 32px;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.c1 {
  -webkit-animation: caVMmK 1.5s ease-in-out infinite;
  animation: caVMmK 1.5s ease-in-out infinite;
  stroke: #007bff;
  stroke-linecap: round;
}

<svg
  class="c0"
  viewBox="-24 -24 48 48"
>
  <circle
    class="c1"
    cx="0"
    cy="0"
    fill="none"
    r="20"
    stroke-width="4"
  />
</svg>
`;
